src/app/Shared/Services/CommandChannel.service.tsx

```diff
@@ -40,12 +40,21 @@ export class CommandChannel {
           if ((jsonResp.dashboardAvailable == true) && (jsonResp.datasourceAvailable == true)) {
             return forkJoin([getDatasourceURL, getDashboardURL])
           } else {
-              return throwError(new Error('ULRs unavailable'));
+            var errMessage = "";
+            if (jsonResp.dashboardAvailable == false) {
+              errMessage = "dashboard URL "
+            }
+            if (jsonResp.datasourceAvailable == false) {
+              errMessage = errMessage.concat(
+                (errMessage == "")? "datasource URL " : ", datasource URL ")
```

Comment:
This error message building is pretty awkward. Most of this ternary is the same, only the comma and additional space are different between the cases, and the `concat` is also a bit unusual.

What about something like this:

```
const missing = [];
if (!dashboardAvailable) {
  missing.push('Dashboard URL');
}
if (!datasourceAvailable) {
  missing.push('datasource URL');
}
const message = missing.join(', ') + ' unavailable.';
return throwError(message);
```

Reply:
yes, much better, should be fixed.